File: modules/spring-core/testdata/java/service/conditionalOnMissingBean/com/app/AppConfiguration2.java
package com.app;

import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class AppConfiguration2 {
    @Bean
    @ConditionalOnMissingBean(TestClass.class)
    public TestClass testClass() {
        return new TestClass();
    }
}

File: modules/spring-core/testdata/kotlin/service/conditionalOnMissingBean/com/app/AppConfiguration2.kt
package com.app;

import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
class AppConfiguration2 {
    @Bean
    @ConditionalOnMissingBean(TestClass::class)
    fun testClass():TestClass {
        return TestClass()
    }
}
